index.bs

```diff
@@ -7467,15 +8838,29 @@ partial  dictionary MLOpSupportLimits {
     1. If [=MLGraphBuilder/validating operand=] with [=this=] and |input| returns false, then [=exception/throw=] a {{TypeError}}.
     1. If any of |sizes|'s [=list/items=] are 0, then [=exception/throw=] a {{TypeError}}.
     1. If |starts|'s [=list/size=] and |sizes|'s [=list/size=] are not both equal to |input|'s [=MLOperand/rank=], then [=exception/throw=] a {{TypeError}}.
-    1. [=list/For each=] |index| in [=the range=] 0 to |input|'s [=MLOperand/rank=], exclusive:
-        1. If |sizes|[|index|] is 0, then [=exception/throw=] a {{TypeError}}.
+    1. Let |strides| be a new [=/list=].
+    1. If |options|.{{MLSliceOptions/strides}} [=map/exists=]:
+        1. Set |strides| to |options|.{{MLSliceOptions/strides}}.
+        1. If |strides|'s [=list/size=] is not equal to |input|'s [=MLOperand/rank=], then [=exception/throw=] a {{TypeError}}.
+    1. Let |inputShape| be |input|'s [=MLOperand/shape=] and |inputRank| be |input|'s [=MLOperand/rank=].
+    1. Let |outputShape| be a new [=/list=].
+    1. [=list/For each=] |index| in [=the range=] 0 to |inputRank|, exclusive:
+        1. Let |inputSize| be |inputShape|[|index|].
+        1. Let |inputSliceSize| be |sizes|[|index|].
+        1. Let |stride| be |strides|[|index|] if it is not empty, or 1 otherwise:
+        1. If |inputSliceSize| is 0, then [=exception/throw=] a {{TypeError}}.
 
             Issue(391): If 0-size dimensions are allowed, revise these steps.
 
-        1. If |starts|[|index|] is greater than or equal to |input|'s [=MLOperand/shape=][|index|], then [=exception/throw=] a {{TypeError}}.
```

Comment:
I changed this line

`1. If |starts|[|index|] is greater than or equal |inputShape|[|index|], then [=exception/throw=] a {{TypeError}}.`

to just `greater than` which prepares it for zero size slices eventually in https://github.com/webmachinelearning/webnn/issues/391, and the check there was redundant anyway as it was already handled by the line immediately below it:

`1. If |starts|[|index|] + |sizes|[|index|] is greater than |inputShape|[|index|], then [=exception/throw=] a {{TypeError}}.`